clouddriver-core/src/main/groovy/com/netflix/spinnaker/clouddriver/orchestration/OperationsService.java

```diff
@@ -0,0 +1,252 @@
+
+                          DescriptionValidator validator =
+                              atomicOperationsRegistry.getAtomicOperationDescriptionValidator(
+                                  DescriptionValidator.getValidatorName(descriptionName),
+                                  provider,
+                                  providerVersion);
+
+                          if (validator == null) {
+                            String operationName =
+                                Optional.ofNullable(description)
+                                    .map(it -> it.getClass().getSimpleName())
+                                    .orElse("UNKNOWN");
+                            log.warn(
+                                "No validator found for operation {} and cloud provider {}",
+                                operationName,
+                                provider);
+                          } else {
+                            // TODO(rz): Assert description is T
+                            validator.validate(descriptions, description, errors);
+                          }
+
+                          allowedAccountValidators.forEach(
+                              it -> {
+                                it.validate(username, allowedAccounts, description, errors);
+                              });
+
+                          // TODO(rz): Assert `description` is T
+                          descriptionAuthorizer.authorize(description, errors);
+
+                          AtomicOperation atomicOperation =
+                              converter.convertOperation(descriptionInput);
+                          if (atomicOperation == null) {
+                            throw new AtomicOperationNotFoundException(descriptionName);
+                          }
+
+                          if (errors.hasErrors()) {
+                            registry
+                                .counter(
+                                    validationErrorsCounterId.withTag(
+                                        "operation", atomicOperation.getClass().getSimpleName()))
+                                .increment();
+                          }
+
+                          return new AtomicOperationBindingResult(atomicOperation, errors);
+                        }))
+        .collect(Collectors.toList());
+  }
+
+  private ProviderVersion getOperationVersion(OperationInput operation) {
+    final String accountName = operation.computeAccountName();
+    if (accountName == null) {
+      log.warn("Unable to get account name from operation: {}", operation);
+    } else {
+      try {
+        AccountCredentials credentials = accountCredentialsRepository.getOne(accountName);
+        return credentials.getProviderVersion();
+      } catch (Exception e) {
+        log.warn("Unable to determine provider version for account {}", accountName, e);
+      }
+    }
+    return ProviderVersion.v1;
+  }
+
+  private static Map processDescriptionInput(
```

Comment:
that's some unusually fancy magic here, can you throw in a comment explaining what is going on for the rest of us?